File: FitnessApp-master/ClientApp/src/Services/WorkoutsService.js
import axios from 'axios';
import Workout from '../Models/Workout';
import Exercise from '../Models/Exercise';
import User from '../Models/User';

export default class WorkoutsService {


    static Exercises(id, callback) {

        axios.get('workouts/exercises/', {
            params: {
                id: id
            }
        })
            .then(function (res) {
                let exList = [];
                if (res.data.length > 0) {
                    for (let i = 0; i < res.data.length; i++) {
                        exList.push(
                            new Exercise(
                                res.data[i].id,
                                res.data[i].name,
                                res.data[i].description)
                        )
                    };
                }
                callback(exList);
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }

    static List(id, callback) {

        axios.get('workouts/list/', {
            params: {
                id: id
            }
        })
            .then(function (res) {
                let workoutslist = [];
                if (res.data.workouts.length > 0) {
                    for (var i = 0; i < res.data.workouts.length; i++) {
                        let workout = res.data.workouts[i];
                        let exercises = workout.exercises;
                        let exerciselist = [];

                        for (let i = 0; i < exercises.length; i++){
                            exerciselist.push(
                                new Exercise(
                                    exercises[i].id,
                                    exercises[i].name,
                                    exercises[i].description)
                            )
                        };

                        let model = new Workout(
                            workout.starttime,
                            workout.endtime,
                            exerciselist,
                            workout.id
                        );
                        workoutslist.push(model);
                    }
                }
                callback(workoutslist);
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }

    static delete(id, userid, callback) {

        axios.delete('workouts/delete/', { params: { id: id } })
            .then(function (res) {

                WorkoutsService.List(userid, callback);
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }

    static add(userid, workout, callback) {
        var data = {
            "userid": userid,
            "starttime": workout.starttime,
            "endtime": workout.endtime,
            "exerciseid": workout.exerciseid
        };
        axios.post('workouts/add/', data)
            .then(function (res) {
                WorkoutsService.List(userid, callback);
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }
}

File: FitnessApp-master/Services/WorkoutsService.cs
using fitnessapp.Models.Workouts;
using fitnessData.AppData;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using System.Linq;

namespace fitnessapp.Services
{
    public class WorkoutsService
    {
        private AppDataDbContext _context;

        public WorkoutsService(AppDataDbContext context)
        {
            _context = context;
        }

        public Workouts_List List(int userid)
        {
            Workouts_List model = new Workouts_List();
            
            if(_context.Workouts.Any(w => w.UserId == userid))
            {
                var workouts = _context.Workouts
                    .Include(w => w.Excercises)
                    .Where(w => w.UserId == userid)
                    .ToList();

                foreach(var workout in workouts)
                {
                    var exerciseList = new List<Excercise>();

                    if (workout.Excercises.Count > 0)
                    {
                        foreach (var e in workout.Excercises)
                        {
                            var b = _context.Excercises.Find(e.ExcerciseId);
                            exerciseList.Add(b);
                        }
                    }

                    model.Workouts.Add(new WorkoutVM(workout.Id, workout.StartTime, workout.EndTime, exerciseList));
                }
            }

            return model;
        }

        public void Add(WorkoutCreateVM model)
        {
            var workout = new Workout(model.userid, model.starttime, model.endtime);

            List<WorkoutExcercise> exList = new List<WorkoutExcercise>();
            var exercise = _context.Excercises.Find(model.exerciseid);
            exList.Add(new WorkoutExcercise(workout, exercise));

            workout.Excercises = exList;

            _context.Add(workout);
            _context.SaveChanges();
        }

        //public void Add(int userid, WorkoutVM model)
        //{
        //    var workout = new Workout(userid, model.starttime, model.endtime);
        //    List<WorkoutExcercise> exList = new List<WorkoutExcercise>();

        //    foreach (var exercise in model.exercises)
        //    {
        //        exList.Add(new WorkoutExcercise(workout, exercise));
        //    }

        //    workout.Excercises = exList;

        //    _context.Add(workout);
        //    _context.SaveChanges();
        //}

        public void Update(WorkoutVM model)
        {
            var workout = _context.Workouts.Find(model.id);
            workout.StartTime = model.starttime;
            workout.EndTime = model.endtime;

            foreach (var exercise in model.exercises)
            {
                if(!workout.Excercises.Any(e => e.ExcerciseId == exercise.Id))
                {
                    var newexercise = _context.Excercises.Find(exercise.Id);

                    workout.Excercises.Add(new WorkoutExcercise(workout, newexercise));
                }
            }

            _context.Update(workout);
            _context.SaveChanges();
        }

        public void Delete(int workoutid)
        {
            var workout = _context.Workouts.Find(workoutid);

            _context.Remove(workout);
            _context.SaveChanges();
        }
    }
}
